docker/worker/testdata/CVE-2016-10046.json

```diff
@@ -0,0 +1,146 @@
+{
+  "id": "CVE-2016-10046",
+  "severity": [
+    {
+      "type": "CVSS_V3",
+      "score": "CVSS:3.0/AV:L/AC:L/PR:N/UI:R/S:U/C:N/I:N/A:H"
+    }
+  ],
+  "details": "Heap-based buffer overflow in the DrawImage function in magick/draw.c in ImageMagick before 6.9.5-5 allows remote attackers to cause a denial of service (application crash) via a crafted image file.",
+  "affected": [
+    {
+      "package": {
+        "name": "imagemagick",
+        "ecosystem": "Debian:11"
+      },
+      "ranges": [
+        {
+          "type": "ECOSYSTEM",
+          "events": [
+            {
+              "introduced": "0"
+            },
+            {
+              "fixed": "8:6.9.6.2+dfsg-2"
+            }
+          ]
+        }
+      ],
+      "ecosystem_specific": {
+        "urgency": "not yet assigned"
+      }
+    },
+    {
+      "package": {
+        "name": "imagemagick",
+        "ecosystem": "Debian:12"
+      },
+      "ranges": [
+        {
+          "type": "ECOSYSTEM",
+          "events": [
+            {
+              "introduced": "0"
+            },
+            {
+              "fixed": "8:6.9.6.2+dfsg-2"
+            }
+          ]
+        }
+      ],
+      "ecosystem_specific": {
+        "urgency": "not yet assigned"
+      }
+    },
+    {
+      "package": {
+        "name": "imagemagick",
+        "ecosystem": "Debian:13"
+      },
+      "ranges": [
+        {
+          "type": "ECOSYSTEM",
+          "events": [
+            {
+              "introduced": "0"
+            },
+            {
+              "fixed": "8:6.9.6.2+dfsg-2"
+            }
+          ]
+        }
+      ],
+      "ecosystem_specific": {
+        "urgency": "not yet assigned"
+      }
+    },
+    {
+      "ranges": [
+        {
+          "type": "GIT",
+          "repo": "https://github.com/imagemagick/imagemagick",
+          "events": [
+            {
+              "introduced": "0"
+            },
+            {
+              "fixed": "989f9f88ea6db09b99d25586e912c921c0da8d3f"
+            }
+          ]
+        },
+        {
+          "type": "GIT",
+          "repo": "https://github.com/imagemagick/imagemagick6",
+          "events": [
+            {
+              "introduced": "0"
+            },
+            {
+              "last_affected": "969a96ed7eea9603bea46492e9116c2ba28da60f"
```

Comment:
Are either of these hashes here meant to be removed? Or is the behaviour that they'll just be ignored for commit expansion purposes but remain in the record?

Reply:
I believe for this particular record, commit expansion was partially successful, hence the results present in the expected diff. It seems like there's a bunch of existing non-crash inducing scenarios that might result in a commit range not expanding successfully, can you advise if all instances of `RepoAnalyzer.get_affected()` returning `None` should bubble up as an exception instead? Or if the GIT range scrubbing behaviour should be contingent on `AnalyzeResult.has_changes` being false?

Reply:
> RepoAnalyzer.get_affected() returning None should bubble up as an exception instead?

I don't think it should bubble up as an exception with its current use. Bubbling this up as an exception would be useful it we needed to take some action (e.g. modify the record), but it seems like this is not the case. 

> GIT range scrubbing behaviour should be contingent on AnalyzeResult.has_changes being false?

This is where my current question is stemming from. It seems like with the current state of the PR, we no longer scrub/discard GIT ranges at all? We just silently ignore these? 

This seems to be the case looking at the testcase -- all the referenced GIT commits are in the expected output diff for the Datastore. 

This is all fine, but we should update the PR description to make this clearer.

Reply:
> This is where my current question is stemming from. It seems like with the current state of the PR, we no longer scrub/discard GIT ranges at all? We just silently ignore these?

Correct.

> This is all fine, but we should update the PR description to make this clearer.

Ack, description adjusted to match current reality.